base/ltcmd.dtx

```diff
@@ -4037,68 +4039,70 @@
 %    \begin{macrocode}
 \cs_new_protected:Npn \@@_arg_to_keyvalue_auxv:nn #1#2
   {
-    \@@_arg_to_keyvalue_loop:nnw {#1} {#2} #2
-      \q_recursion_tail \q_recursion_stop
+    \@@_arg_to_keyvalue_loop:w #2
+      \q_recursion_tail \q_recursion_stop {#1} {#2}
```

Comment:
I understand your thinking here, but this does make debugging quite a bit more 'interesting' as you have to remember what's on the stack all the time. (Not saying you should change it, just an observation.)

Reply:
I thought this one wasn't too convoluted (the final two arguments are just used at the very end), and it would spare a few cycles, since this is kind of “inner loop”.  I'll add something to the documentation to make it easier to follow in the future.